tests/beaker_tests/lib/utilitylib.rb

```diff
@@ -137,7 +137,7 @@ def raise_skip_exception(message, testcase)
 
 # Full command string for puppet agent
 def puppet_agent_cmd
-  PUPPET_BINPATH + 'agent -t'
+  PUPPET_BINPATH + 'agent -t --trace'
```

Comment:
is this change for dumping the trace when there is failure always? will there be any performance implication ?

Reply:
I did not see any noticeable performance change.  This is a huge benefit for debugging when a error occurs so I decided to add it.  It will only dump a trace on error, not failure.

Reply:
Yup, this should be okay from what I remember.